### ISSUE TITLE
Resource only topicrefs considered duplicates by key processing

### ISSUE THREAD

**Issue by jelovirt** [dita-ot/dita-ot#2304]
For map

```xml
<topicref href="foo.dita" processing-role="resource-only"/>
<topicref href="foo.dita" />
```

The latter topicref is considered a duplicate by key processing and is renamed to `foo-1.dita`. Resource only topicrefs should not be counted when considering duplicates.

DITA-OT 2.2

**Comment by eerohele** [comment]
This happens to fix #2134 because the test files in that issue use `<keydef>`, which has implicit `@processing-role="resource-only"`, but IIRC #2134 still occurs if the key-defining topicref has a normal processing role.

**Comment by eerohele** [comment]
No, I remembered wrong, I reread what I wrote in #2134 and ran some tests and it seems this resolves #2134 as well. If something additional surfaces, we can open a new issue.

(This doesn't resolve #2242, though.)
